### ISSUE TITLE
It should be possible to restrict a plugin to be a descendant of another

### ISSUE THREAD

**Issue by mkoistinen** [django-cms/django-cms#4939]
This is different than simple parent-restrictions in that there may be intermediate plugins. Example:

Form Plugin
- Style Plugin
  - Field Plugin

Field plugins should be restricted to be a descendant of Form, even if its not the direct parent.

**Comment by fsbraun** [comment]
I agree: https://github.com/django-cms/djangocms-form-builder/issues/26

**Comment by fsbraun** [comment]
With #8181 merged, django CMS allows to implement this functionality to a reasonable extend.
